packages/core/src/system/Collapse/types.ts

```diff
@@ -0,0 +1,44 @@
+import { ReactNode } from 'react';
+import { BaseProps, ComponentPropsWithHTML, TransitionProps, WithClassesAndStyles } from '@core';
+import { CollapseStyleKeys } from './styles';
+
+export type CollapseOrientation = 'horizontal' | 'vertical'
+
+export interface CollapsePropsWithoutHtml extends
+    BaseProps,
+    WithClassesAndStyles<CollapseStyleKeys>
+{
+    /** ReactNode, который должен быть свернут */
+    children: ReactNode;
+
+    /** Если true, то элемент развёрнут */
+    open: boolean;
+
+    /** Временные функции появления и исчезновения
+     *
+     * @default false */
+    easing?: TransitionProps['easing'];
+
+    /** Направление скрытия/раскрытия
+     *
+     * @default vertical */
+    orientation?: CollapseOrientation;
+
+    /** Длительность анимации
+     *
+     * @default false */
+    timeout?: TransitionProps['timeout'];
+
+    /** Ширина (при orientation = horizontal) или высота (при orientation = vertical) контейнера,
+     * когда он всёрнут
+     *
+     * @default 0 */
+    collapsedSize?: number | string;
+
+    /** Свойства используемого компонента Transition
+     *
+     * @default false */
+    transitionProps?: Omit<TransitionProps, Exclude<keyof CollapsePropsWithoutHtml, 'transitionProps'>>;
```

Comment:
при передаче пропсов в другой компонент, лучше имя компонента писать с большой буквы, чтобы было понятно, что речь именно о внутреннем компоненте, а не просто похожее название пропса

```suggestion
    TransitionProps?: Omit<TransitionProps, Exclude<keyof CollapsePropsWithoutHtml, 'transitionProps'>>;
```